Actors library alpha changelog (#23206)
cloudflare/cloudflare-docs: src/content/changelog/durable-objects/2025-06-25-actors-package-alpha.mdx

---
title: @cloudflare/actors library: SDK for Durable Objects in alpha
description: @cloudflare/actors library with Durable Objects helpers and patterns.
products:
  - durable-objects
  - workers
date: 2025-06-25T06:00:00Z
---

> [@cloudflare/actors](https://www.npmjs.com/package/@cloudflare/actors) package is in alpha and provides a SDK for Durable Objects, in addition to Durable Objects [Workers API](https://developers.cloudflare.com/durable-objects/api/base/). "Actors" originates from the [actor programming model](/durable-objects/what-are-durable-objects/#actor-programming-model), which is a close analogy for Durable Objects.

Reviewer: ```suggestion
The new [@cloudflare/actors](https://www.npmjs.com/package/@cloudflare/actors) library now is in beta!

The `@cloudflare/actors` library is a new SDK for Durable Objects, and provides a powerful set of abstractions for building multiplayer, collaborative and/or WebSockets-based applications on top of Durable Objects.

The name "actors" originates from the [actor programming model](/durable-objects/what-are-durable-objects/#actor-programming-model), which closely ties to how Durable Objects are modelled.
```

package -> library
alpha -> beta?
communicate value + "why"

Author: done with some rewording, websocket focus to come in Q3

> For alpha, `/actors` library includes:

Reviewer: ```suggestion
The `@cloudflare/actors` library includes:
```

Storage and alarm helper methods can be combined with [any Javascript class](https://github.com/cloudflare/actors?tab=readme-ov-file#storage--alarms-with-durableobject-class) that defines your Durable Object, i.e, ones that extend `DurableObject` including the `Actor` class.

```js
import { Storage } from "@cloudflare/actors/storage";

export class ChatRoom extends DurableObject<Env> {
    storage: Storage;
    
    constructor(ctx: DurableObjectState, env: Env) {
        super(ctx, env)
        this.storage = new Storage(ctx.storage);
        this.storage.migrations = [{
            idMonotonicInc: 1,
            description: "Create users table",
            sql: "CREATE TABLE IF NOT EXISTS users (id INTEGER PRIMARY KEY)"
        }]
    }
    async fetch(request: Request): Promise<Response> {
        // Run migrations before executing SQL query
        await this.storage.runMigrations();

        // Query with SQL template
        let userId = new URL(request.url).searchParams.get("userId");
        const query = this.storage.sql`SELECT * FROM users WHERE id = ${userId};`
        return new Response(`${JSON.stringify(query)}`);
    }
}
```

> `/actors` library introduces the `Actor` class pattern. `Actor` lets you access Durable Objects without writing the Worker that communicates with your Durable Object (the Worker is created for you). By default, requests are routed to a Durable Object named "default". 

Reviewer: ```suggestion
`@cloudflare/actors` library introduces the `Actor` class pattern. `Actor` lets you access Durable Objects without writing the Worker that communicates with your Durable Object (the Worker is created for you). By default, requests are routed to a Durable Object named "default". 

Use the full package name so it's clear throughout + what a user will type.
```
